processor/metricstarttimeprocessor/internal/datapointstorage/timeseries_map.go

```diff
@@ -102,6 +102,82 @@ func (tsm *TimeseriesMap) GC() {
 	tsm.Mark = false
 }
 
+func (tsi *TimeseriesInfo) IsResetHistogram(h pmetric.HistogramDataPoint) bool {
+	if h.Count() != 0 && tsi.Histogram.Count() != 0 && h.Count() < tsi.Histogram.Count() {
+		return true
+	}
+	if h.Sum() != 0 && tsi.Histogram.Sum() != 0 && h.Sum() < tsi.Histogram.Sum() {
+		return true
+	}
+
+	// Guard against bucket boundaries changes.
+	refBounds := tsi.Histogram.ExplicitBounds().AsRaw()
+	hBounds := h.ExplicitBounds().AsRaw()
+	if len(refBounds) != len(hBounds) {
+		return true
+	}
+	for i := range len(refBounds) {
+		if hBounds[i] != refBounds[i] {
+			return true
+		}
+	}
+
+	// We need to check individual buckets to make sure the counts are all increasing.
+	if tsi.Histogram.BucketCounts().Len() != h.BucketCounts().Len() {
+		return true
+	}
+	for i := range tsi.Histogram.BucketCounts().Len() {
+		if h.BucketCounts().At(i) < tsi.Histogram.BucketCounts().At(i) {
+			return true
+		}
+	}
+	return false
+}
+
+func (tsi *TimeseriesInfo) IsResetExponentialHistogram(eh pmetric.ExponentialHistogramDataPoint) bool {
+	// Same as the histogram implementation
+	if eh.Count() != 0 && tsi.ExponentialHistogram.Count() != 0 && eh.Count() < tsi.ExponentialHistogram.Count() {
```

Comment:
Same question as non-exponential histograms, wouldn't a new 0 count data point count as a reset?

Reply:
Thats a good point, the zero comparison isn't needed. As long as the value decrements, it should count as a reset.